internal/controllers/topology/cluster/patches/template.go

```diff
@@ -135,15 +136,11 @@ func getRequestItem(req *runtimehooksv1.GeneratePatchesRequest, holderKind, hold
 // bytesToUnstructured provides a utility method that converts a (JSON) byte array into an Unstructured object.
 func bytesToUnstructured(b []byte) (*unstructured.Unstructured, error) {
 	// Unmarshal the JSON.
-	var m map[string]interface{}
-	if err := json.Unmarshal(b, &m); err != nil {
+	u := &unstructured.Unstructured{}
+	if _, _, err := serializer.NewCodecFactory(nil).UniversalDeserializer().Decode(b, nil, u); err != nil {
```

Comment:
I'm not sure but I think creating a new codec factory and a new deserializer for every bytesToUnstructured is too much.

Let's use a package global variable for the deserializer that we initialize with `init`. I think this should be okay, adding it to the engine and passing it through might be too much effort.

Reply:
\* don't need init works inline in the var declaration